addons/html_editor/controllers/main.py

```diff
@@ -570,15 +570,20 @@ def link_preview_metadata_internal(self, preview_url):
 
             record_id = int(words.pop())
             action_name = words.pop()
-            action = Actions.sudo().search([('path', '=', action_name)])
-            if not action:
-                return {'error_msg': _("Action %s not found, link preview is not available, please check your url is correct", action_name)}
-            action_type = action.type
-            if action_type != 'ir.actions.act_window':
-                return {'other_error_msg': _("Action %s is not a window action, link preview is not available", action_name)}
-            action = request.env[action_type].browse(action.id)
-
-            model = request.env[action.res_model].with_context(context)
+            if action_name in request.env:
+                """if path format is `odoo/<model>/<record_id>` so we use `action_name` as model name"""
+                model = request.env[action_name].with_context(context)
```

Comment:
Here since it's actually a model's name we are using, it better to use `model_name` instead. Also, we cover another possibility of model name in the URL which starts with `m-`
```suggestion
                model_name = action_name.removeprefix('m-')
                model = request.env[model_name].with_context(context)
```